sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -585,6 +586,74 @@ public String getId() {
+     *
+     * @param <TContext> The context for the bulk processing.
+     * @param operations Flux of operation which will be executed by this container.
+     *
+     * @return A Flux of {@link CosmosBulkOperationResponse} which contains operation and it's response or exception.
+     * <p>
+     *     To create a operation which can be executed here, use {@link BulkOperations}. For eg.
+     *     for a upsert operation use {@link BulkOperations#newUpsertItemOperation(Object, PartitionKey)}
+     * </p>
+     * <p>
+     *     We can get the corresponding operation using {@link CosmosBulkOperationResponse#getOperation()} and
+     *     it's response using {@link CosmosBulkOperationResponse#getResponse()}. If the operation was executed
+     *     successfully, the value returned by {@link CosmosBulkItemResponse#isSuccessStatusCode()} will be true. To get
+     *     actual status use {@link CosmosBulkItemResponse#getStatusCode()}.
+     * </p>
+     * To check if the operation had any exception, use {@link CosmosBulkOperationResponse#getThrowable()} to
+     * get the exception.
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <TContext> Flux<CosmosBulkOperationResponse<TContext>> processBulkOperations(
+        Flux<CosmosItemOperation> operations) {
+
+        return this.processBulkOperations(operations, new BulkProcessingOptions<>());
+    }
+
+    /**
+     * Executes flux of operations in Bulk.
+     *
+     * @param <TContext> The context for the bulk processing.
+     *
+     * @param operations Flux of operation which will be executed by this container.
+     * @param bulkOptions Options that apply for this Bulk request which specifies options regarding execution like
+     *                    concurrency, batching size, interval and context.
+     *
+     * @return A Flux of {@link CosmosBulkOperationResponse} which contains operation and it's response or exception.
+     * <p>
+     *     To create a operation which can be executed here, use {@link BulkOperations}. For eg.
+     *     for a upsert operation use {@link BulkOperations#newUpsertItemOperation(Object, PartitionKey)}
+     * </p>
+     * <p>
+     *     We can get the corresponding operation using {@link CosmosBulkOperationResponse#getOperation()} and
+     *     it's response using {@link CosmosBulkOperationResponse#getResponse()}. If the operation was executed
+     *     successfully, the value returned by {@link CosmosBulkItemResponse#isSuccessStatusCode()} will be true. To get
+     *     actual status use {@link CosmosBulkItemResponse#getStatusCode()}.
+     * </p>
+     * To check if the operation had any exception, use {@link CosmosBulkOperationResponse#getThrowable()} to
+     * get the exception.
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <TContext> Flux<CosmosBulkOperationResponse<TContext>> processBulkOperations(
+        Flux<CosmosItemOperation> operations,
+        BulkProcessingOptions<TContext> bulkOptions) {
```

Comment:
See my previous comment on TContext... I don't think we need it, so we might just remove this and keep the BulkProcessingOptions simpler.

Reply:
we'll need it - that is the way how the caller can pass through state needed to update bookmarks etc. We discussed this when we iterated over it in the public surface area discussions. Please leave it for now - we can add samples and then iterate if really needed.